kernel/kernel.h

```diff
@@ -78,6 +78,7 @@ void intr_irq_handler(uint64_t irq);
 
 /* mem.c: low-level page alloc routines */
 void mem_init(void);
+void *alloc_chunk_4K(size_t size);
```

Comment:
Can you add a comment there saying that this function should only be called before `solo5_app_main`? as doing so will most likely mess with the unikernel heap. In fact, it would be nice to add some assert or some mechanism that ensures that.